webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/RuleApi.scala

```diff
@@ -739,9 +739,16 @@ class RuleApiService14 (
                   , params.reason
                 )
       id       <- workflow.startWorkflow(cr, actor, params.reason)
+      directiveLib <- readDirectives.getFullDirectiveLibrary().toBox
+      groupLib <- readGroup.getFullGroupLibrary().toBox
+      nodesLib <- readNodes.getAll().toBox
+      globalMode  <- getGlobalPolicyMode().toBox
+
     } yield {
+      val status = getRuleApplicationStatus(change.newRule, groupLib, directiveLib, nodesLib, globalMode)
```

Comment:
we should really keep the rule application status somewhere in cache, it does not make any sens to compute it each time when we can track each things that can make it change. 

Perhaps a first step is to extract a method that does the status just from the ruleid so that it is easier to centralize a cache (not for now, but for 7.1)